### ISSUE TITLE
update index queue if a SOLR_RELATION record changes?

### ISSUE THREAD

**Issue by martinhummer** [TYPO3-Solr/ext-solr#131]
is it possible at the moment that the index queue gets automatically updated if a table (record) which is defined as SOLR_RELATION gets updated?

i'm aware of the RecordMonitor.php but can't find any solution there.

thanks!

**Comment by Mabahe** [comment]
Not implemented at the moment. I need a similar feature soon and it is planned to provide a patch. But I personally would not depend this on the SOLR_RELATION, because SOLR_RELATION is not the only way to push related data to the index. I plan to check TCA relations to records where also an index queue is defined for.

**Comment by martinhummer** [comment]
yes, i also think that's the way to go.
if you need some help or a tester, let me know.

**Comment by timoschmidt** [comment]
@Mabahe 
@martinhummer 

Sounds good! By now i am not so deep in this topic, but it think this is a usefull feature. And i have also some ideas in mind what could be usefull. e.g. using the sys_ref_index. When i can support somewhere just ask me!

**Comment by timohund** [comment]
Also mentioned here: https://forge.typo3.org/issues/44586

I think we need to have a concept to detect what records need to be reIndexed when a record is processed by the TYPO3 DataHandler. Togehter with the sys_refindex & some configuration we should be able to do this.

**Comment by timohund** [comment]
A while ago i created a concept for an extension to map DataHandler and sys_refindex information to an event model:

https://github.com/timohund/ext-typo3-eventapi

I think this might take some time to implement but would be nice to have in the TYPO3 core.

**Comment by neufeind** [comment]
In case TYPO3-core takes care of clears, can we make use of that? Like if for example TYPO3 knows a certain record is changed so it needs to invalidate a number of pages where that record is displayed (potentially by some plugin or so). That way it would trigger clears of that page and we might know we'll need to reindex that "related" record as well?

For other kinds of changes we discussed several times in TYPO3-core as well as EXT:solr who is responsible. I'm thinking of changes to a setting on a page higher up that results in changes for (indexed) pages on the levels below. Part of that is also changing visibility (or even deleting?) for a whole subtree, where currently the records might still stay untouched in the index.
At that point it was open whether TYPO3-core should handle such recursive notifications (for subpages, ...) or whether EXT:solr should invent their own solution (and imho possibly miss a certain special-case). Personally I'd be in favor for having such (recursive) notifications done in the core and being available to other extensions.

**Comment by timohund** [comment]
@neufeind i would agree on that. I think currently it is very hard (or nearly impossible) to detect the complete impact of a change to a content element or a record. But i would say it should be possible with:

- The existing hooks and data & signal and slots
- The sysref index
- Some additional configuration

To allow to detect more changes and allow to implement an event model. But as i said this is quite some effort to implement.
